streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -142,6 +143,16 @@ void initializeNewTasks() {
                 transitionToRunning(task);
                 resume.addAll(task.partitions());
                 it.remove();
+            } else {
+                if (log.isTraceEnabled()) {
+                    final HashSet<TopicPartition> outstandingPartitions = new HashSet<>(task.changelogPartitions());
+                    outstandingPartitions.removeAll(restoredPartitions);
```

Comment:
Why is this statement wrapped under log.isTraceEnabled() condition ?

Reply:
So we don't bother creating new HashSets etc if it isn't going to be logged